mobile/examples/phi-3/ios/LocalLLM/LocalLLM/GenAIGenerator.mm

```diff
@@ -5,45 +5,109 @@
 #include "LocalLLM-Swift.h"
 #include "ort_genai.h"
 #include "ort_genai_c.h"
-
+#include <chrono>
 
 @implementation GenAIGenerator
 
-+ (void)generate:(nonnull NSString*)input_user_question {
-  NSString* llmPath = [[NSBundle mainBundle] resourcePath];
-  const char* modelPath = llmPath.cString;
-
-  auto model = OgaModel::Create(modelPath);
-  auto tokenizer = OgaTokenizer::Create(*model);
-
-  NSString* promptString = [NSString stringWithFormat:@"<|user|>\n%@<|end|>\n<|assistant|>", input_user_question];
-  const char* prompt = [promptString UTF8String];
-
-  auto sequences = OgaSequences::Create();
-  tokenizer->Encode(prompt, *sequences);
+typedef std::chrono::high_resolution_clock Clock;
+typedef std::chrono::time_point<Clock> TimePoint;
 
-  auto params = OgaGeneratorParams::Create(*model);
-  params->SetSearchOption("max_length", 200);
-  params->SetInputSequences(*sequences);
-
-  // Streaming Output to generate token by token
-  auto tokenizer_stream = OgaTokenizerStream::Create(*tokenizer);
-
-  auto generator = OgaGenerator::Create(*model, *params);
++ (void)generate:(nonnull NSString*)input_user_question {
+    NSLog(@"Starting token generation...");
+    
+    NSString* llmPath = [[NSBundle mainBundle] resourcePath];
+    const char* modelPath = llmPath.cString;
+    
+    // Log model creation
+    NSLog(@"Creating model ...");
+    auto model = OgaModel::Create(modelPath);
+    if (!model) {
+        NSLog(@"Failed to create model.");
+        return;
+    }
+    
+    NSLog(@"Creating tokenizer...");
+    auto tokenizer = OgaTokenizer::Create(*model);
+    if (!tokenizer) {
+        NSLog(@"Failed to create tokenizer.");
+        return;
+    }
```

Comment:
The tokenizer could also be created once and re-used as I believe it's tied to the model not the prompt so can be re-used.